hw_diag/tests/test_views_diagnostics.py

```diff
@@ -86,3 +87,27 @@ def test_version_endpoint(self):
         self.assertEqual(reply['firmware_version'], '1337.13.37')
         self.assertEqual(reply['diagnostics_version'], 'aabbffe')
         self.assertEqual(reply['firmware_short_hash'], '0011223')
+
+    @patch('hw_diag.diagnostics.add_gateway_txn_diagnostic.AddGatewayTxnDiagnostic.perform_test')
+    def test_add_gateway_txn_success(self, mock):
```

Comment:
We haven't really talked about this as a team, but for API endpoints, I think we should try to implement integration tests when possible. In this case, it would mean not mocking the diagnostic. You don't need to change this, but a thought for the future.

Reply:
Good advice :+1: ! As a best practice, I will add both of the unit test(for ex, mocking the diagnostic) and the integration test(for ex, not mocking the diagnostic or mocking the lower level such as miner API) in the future.

Reply:
Once testing for hm-dashboard is available we could setup integration tests between hm-diag <-> HPT <-> hm-dashboard.